Your_first_ASDF_converter.ipynb

```diff
@@ -0,0 +1,554 @@
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "09ee1385",
+   "metadata": {},
+   "source": [
+    "Next we create the file that contains the converter code"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "aa345e79",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%writefile src/mfconverter/converter.py\n",
+    "from asdf.extension import Converter\n",
+    "from .photo_id import PhotoID\n",
+    "\n",
+    "class PhotoIDConverter(Converter):\n",
+    "    tags = [\"asdf://stsci.edu/example-project/photo_id-*\"]\n",
+    "    types = [\"mfconverter.photo_id.PhotoID\"]\n",
+    "    # The above registers the tag that the converter is used for, as well as\n",
+    "    # associating the class that the converter is used for.\n",
+    "    \n",
+    "    # This method converts from the Python object to yaml\n",
+    "    def to_yaml_tree(self, obj, tags, ctx):\n",
+    "        # The yaml conversion expects a dictionary returned\n",
+    "        node = {}\n",
+    "        node['first_name'] = obj.first_name\n",
+    "        node['last_name'] = obj.last_name\n",
+    "        node['photo'] = obj.photo\n",
+    "        return node\n",
+    "    \n",
+    "    # This method converts from yaml to the Python object\n",
+    "    def from_yaml_tree(self, node, tag, ctx):\n",
+    "        return PhotoID(node['last_name'],\n",
+    "                       node['first_name'],\n",
+    "                       node['photo'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "55ca30cd",
+   "metadata": {},
+   "source": [
+    "In the above example, the class attribute `types` is a list since it is\n",
+    "possible for multiple types to use the same converter. Secondly, it is\n",
+    "possible to supply the type as either a string (as done above) as the \n",
+    "actual class itself. The former is preferred for peformance reasons as\n",
+    "using the actual class itself forces the ASDF package to import the\n",
+    "module containing the class, even if it is never used in the program\n",
+    "using ASDF.\n",
+    "\n",
+    "The wildcard for the tag entry indicates that this converter will work\n",
+    "with all versions of the tag. It isn't strictly needed here, but generally \n",
+    "a good practice if one wants the converter code to handle multiple versions.\n",
+    "(How to handle versioning is a topic in its own right and not covered here)\n",
+    "\n",
+    "We need to create a module for the entry point handling"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f9c03910",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%writefile src/mfconverter/extensions.py\n",
+    "from asdf.extension import Extension\n",
+    "from .converter import PhotoIDConverter\n",
+    "\n",
+    "class MFExtension(Extension):\n",
+    "    extension_uri = \"asdf://stsci.edu:example-project/photo_id-1.0.0\"\n",
```

Comment:
It's unusual to have an http-style URI with a colon after the authority.  I recommend a forward slash here.

The docs also recommend the /extensions/ path, which we should change if we prefer this format:

https://asdf.readthedocs.io/en/latest/asdf/extending/uris.html#extensions

Reply:
I thought I had gotten rid of that. But apparently not.